DependencyInjection/DoctrineExtension.php

```diff
@@ -85,6 +87,14 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
     {
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
         $loader->load('dbal.xml');
+        $chainLogger     = $container->getDefinition('doctrine.dbal.logger.chain');
+        $profilingLogger = new Reference('doctrine.dbal.logger');
+        if (! method_exists(SQLParserUtils::class, 'getPositionalPlaceholderPositions') && method_exists(LoggerChain::class, 'addLogger')) {
+            // doctrine/dbal < 2.10.0
+            $chainLogger->addMethodCall('addLogger', [$profilingLogger]);
+        } else {
+            $chainLogger->addArgument([$profilingLogger]);
+        }
```

Comment:
@stof here is the replacement you asked for in https://github.com/doctrine/DoctrineBundle/pull/1263/files#r524450538 (I think it's equivalent)

Reply:
this is not equivalent. As you need to pass an array of loggers to the constructor rather than multiple calls to `addLogger`, you need to build an array containing them all.

Reply:
But there is only one logger before this PR, right?

Reply:
What are you suggesting exactly? This?

```
            $chainLogger->replaceArgument(
                $loggers,
                array_merge($chainLogger->getArgument($loggers), $logger)
            );
```

Reply:
no. Cases where the chain logger are used are precisely cases where we have 2 loggers:

- `doctrine.dbal.logger` logging to your app PSR-3 logger (generally Monolog)
- the profiling logger which logs query executions for the profiler

Your suggestion won't work if the reference to `doctrine.dbal.logger` is added to the constructor of the `doctrine.dbal.logger.chain` abstract definition, because the child definition won't have anything yet for the argument. Instead, you should just use `[$logger, $profilingLogger]` as the value of the argument

Reply:
In the existing code, the call done in XML is done on the abstract service definition which is extended by the definitions in each of the DBAL connections.

so after resolving the definition inheritance, each chain logger will have 2 `addLogger` calls:

- the one inherited from the abstract definition for `doctrine.dbal.logger`
- the one set on themselves for the connection-specific profiling logger.